ckanext/geodatagov/cli.py

```diff
@@ -290,18 +290,43 @@ def index_for(_type):
         return NoopSearchIndex()
 
 
-def get_all_entity_ids_and_date(max_results: int = 1000):
+def get_all_entity_ids_date_hoid():
     """
     Return a list of the IDs and metadata_modified of all indexed packages.
     """
     query = "*:*"
     fq = '+site_id:"%s" ' % config.get("ckan.site_id")
     fq += "+state:active "
+    fq += "+type:dataset "
 
+    ret_all = []
+
+    start = 0
+    page_size = 10000
     conn = make_connection()
-    data = conn.search(query, fq=fq, rows=max_results, fl="id, metadata_modified")
 
-    return [(r.get("id"), r.get("metadata_modified")) for r in data.docs]
+    log.info(f"Now loading SOLR packages using page size {page_size}...")
+
+    while True:
+        log.info(f"loading packages starting from {start}")
+        data = conn.search(query, fq=fq, start=start, rows=page_size, fl="id, metadata_modified, validated_data_dict")
+
+        if not data:
+            break
+
+        for r in data.docs:
+            harvest_object_id = None
+            data_dict = json.loads(r.get("validated_data_dict"))
+            for extra in data_dict.get("extras", []):
+                if extra["key"] == "harvest_object_id":
+                    harvest_object_id = extra["value"]
+                    break
```

Comment:
Maybe we should document that this function is what makes the entire function memory-expensive and is the cause for the "paging" algorithm.  I would put it in the function header block.

Reply:
comments added.